modules/infra/submodules/storage/datasync.tf

```diff
@@ -0,0 +1,104 @@
+locals {
+  create_ds = var.storage.netapp.migrate_from_efs.datasync.enabled
+  datasync_security_group_rules = local.create_ds ? {
+    efs = {
+      protocol                 = "-1"
+      from_port                = 0
+      to_port                  = 0
+      description              = "EFS allow all traffic to DataSync"
+      source_security_group_id = aws_security_group.datasync[0].id
+      security_group_id        = aws_security_group.efs[0].id
+    }
+    netapp = {
+      protocol                 = "-1"
+      from_port                = 0
+      to_port                  = 0
+      description              = "Netapp allow all traffic to DataSync"
+      source_security_group_id = aws_security_group.datasync[0].id
+      security_group_id        = aws_security_group.netapp[0].id
+    }
+  } : {}
+
+}
+
+resource "aws_security_group" "datasync" {
+  count  = local.create_ds ? 1 : 0
+  name   = "${var.deploy_id}-datasync"
+  vpc_id = var.network_info.vpc_id
+
+  description = "Datasync security group"
+
+  tags = {
+    "Name" = "${var.deploy_id}-datasync"
+  }
+}
+
+resource "aws_security_group_rule" "datasync_egress" {
+  for_each = local.create_ds ? local.datasync_security_group_rules : {}
+
+  security_group_id        = each.value.source_security_group_id
+  protocol                 = each.value.protocol
+  from_port                = each.value.from_port
+  to_port                  = each.value.to_port
+  type                     = "egress"
+  description              = each.value.description
+  source_security_group_id = each.value.security_group_id
+}
+
+resource "aws_security_group_rule" "datasync_ingress" {
+  for_each = local.create_ds ? local.datasync_security_group_rules : {}
+
+  security_group_id        = each.value.security_group_id
+  protocol                 = each.value.protocol
+  from_port                = each.value.from_port
+  to_port                  = each.value.to_port
+  type                     = "ingress"
+  description              = each.value.description
+  source_security_group_id = each.value.source_security_group_id
+}
+
+data "aws_subnet" "ds" {
+  count = local.create_ds ? 1 : 0
+  id    = var.network_info.subnets.private[0].subnet_id
+}
+
+resource "aws_datasync_location_efs" "this" {
```

Comment:
does this need [in_transit_encryption](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/datasync_location_efs#in_transit_encryption)?